lib/src/io/node.dart

```diff
@@ -76,15 +79,15 @@ Future<String> readStdin() async {
   });
   // Node defaults all buffers to 'utf8'.
   var sink = utf8.decoder.startChunkedConversion(innerSink);
-  process.stdin.on('data', allowInterop(([Object? chunk]) {
+  process?.stdin.on('data', allowInterop(([Object? chunk]) {
```

Comment:
Currently, if this is called in a context without access to `process`, it'll return a future that silently never completes which could make debugging confusing. It's probably better to check whether `process` is available eagerly and throw an error if it isn't. For example:

```dart
// Reassigning to a variable is necessary for null analysis to work for now.
var process_ = process;
if (process_ == null) {
  throw UnsupportedError("readStdin() is only supported on Node.js");
}

// ...
```

We should probably do something similar for all the Node-specific methods here. That way, if we end up accidentally invoking them from browser-available contexts later on down the line, we'll get helpful error messages.

Reply:
Fixed in 2dcf8cdebf48242408025fc9297e68b8c41f7deb, and also added checks to all functions